test/unit/test_dwarf.py

```diff
@@ -14,3 +15,14 @@ def test_no_crash(self):
             args = [os.path.join(path, name)] + \
                    ['-g', '--dwarfdump', '--roundtrip', '--dwarfdump']
             shared.run_process(shared.WASM_OPT + args, capture_output=True)
+
+    def test_dwarf_incompatibility(self):
```

Comment:
This could make a really simple lit test:

```
;; RUN: wasm-opt -g <path to cubescript.wasm> --flatten 2>&1 | filecheck %s --check-prefix WARN
;; RUN: wasm-opt -g <path to cubescript.wasm> --metrics 2>&1 | filecheck %s --check-prefix OK

;; WARN: not fully compatible with DWARF
;; OK-NOT: not fully compatible with DWARF

Reply:
Ok, happy to try to join the lit bandwagon...

Do we have instructions anywhere for how to set things up locally? `./check.py  lit` gives me errors,

```
parse-error.wast.script: line 1: filecheck: command not found
```

I have done `pip install filecheck` (that worked for `lit` in the past iirc) but that appears to not be enough. Do I need a build of LLVM or something like that?

Reply:
`pip3 install -r dev-requirements.txt` should get you up and running. Thanks to @sbc100 for suggesting that we use such a file.

Reply:
(looks like it's `requirements-dev.txt`?)

Ok, I ran `pip3 install -r requirements-dev.txt` and it succeeded, but I still get `filecheck: command not found`.

I am not very familiar with `pip`. It did not install anything called `*filecheck*` under my current dir. Does it install to a global location?

Reply:
Pip install (when run without --system) install things into a user-local PATH.  On linux (at least on my machine) this is `$HOME/.local/bin`.   I think the expectation is that this would be in your PATH, or that you would add it to your path.   I'm having trouble finding docs on best practices around how to ensure its added.

Reply:
Thanks!

Indeed that location is not in my path. Adding it fixes things...

Reply:
Ok, looks like I need to add it as a `.wast` file, even though it's not a wast. Now it passes.

However, I am a little unsure this is better. It has
```
;; WARN: not fully compatible with DWARF
;; OK-NOT: not fully compatible with DWARF
```
The first line is robust in that a typo will make the test fail. The second line is brittle, however. In the python version a single string is used to avoid that risk, but it must be duplicated in the lit...?

If that's true then I think I'd prefer to keep it as python. But I don't feel strongly?

Reply:
Yes, the line has to be duplicated in the lit test, but on the whole I still prefer the lit test over the Python test due to the relative verbosity of the latter. If you want to use a different file extension, you can add it to the lit config here: https://github.com/WebAssembly/binaryen/blob/main/test/lit/lit.cfg.py#L6.

Reply:
Ok, I don't feel strongly so let's make it lit. Pushed now, PTAL.